arch/arm/core/cortex_a_r/exc.S

```diff
@@ -47,7 +47,11 @@ GTEXT(z_arm_data_abort)
 	 * construct an exception stack frame.
 	 */
 	srsdb sp!, #\mode
+#if defined(CONFIG_ASSEMBLER_ISA_THUMB2)
+	stmfd sp, {r0-r3, r12, lr}
```

Comment:
TBH, I am actually a bit confused about this since the ^ is not related to THUMB2?

Reply:
Oh, is that because THUMB2 doesn't support the ^ option? I am thinking there is any way that we can also get rid of the guard too?

Reply:
that's right, is not supported. I will check if it's possible

Reply:
hi @povergoing. The `^` seems to be redundant in all the cases used here. When in ARM mode, according to https://developer.arm.com/documentation/dui0489/c/arm-and-thumb-instructions/memory-access-instructions/ldm-and-stm and inspecting the registers e.g. during undef exception handler, there seems to be no effect. I'd say is safe to drop the `^` for both ARM and Thumb2 modes.
But since I'm not so sure why is there in the first place, I want to confirm with you / @microbuilder before doing the change.